cli/lsp/tsc.rs

```diff
@@ -4341,7 +4341,9 @@ impl TscSpecifierMap {
     if let Some(specifier) = self.normalized_specifiers.get(original) {
       return Ok(specifier.clone());
     }
-    let specifier_str = original.replace(".d.ts.d.ts", ".d.ts");
+    let specifier_str = original
+      .replace(".d.ts.d.ts", ".d.ts")
+      .replace("$node_modules", "node_modules");
```

Comment:
Huh, I kept trying this as a hail mary for some other bugs but could never manifest a case where tsc was constructing paths bypassing `op_resolve()` (which typically stores mappings for every specifier when denormalizing its results). I guess it's happening at `host.resolveTypeReferenceDirectiveReferences()`.

Nice find!